File: src/ClientApp/src/app/pages/clients/view-models/client-timeline.view-model.ts
export interface ClientTimelineViewModel {
    entityType: string;
    entityId: string;
    activityType: string;
    title: string;
    description?: string | null;
    activityDate: string;
    createdByUserName?: string | null;
    isSystemGenerated: boolean;
    referenceEntity?: string | null;
}

export interface ClientTimelineResponseViewModel {
    items: ClientTimelineViewModel[];
    pageNumber: number;
    pageSize: number;
    totalCount: number;
    totalPages: number;
    activityTypes: string[];
}

File: src/API/src/Modules/Clients/Clients.Application/ViewModels/ClientTimelineViewModel.cs
namespace Clients.Application.ViewModels
{
    public class ClientTimelineViewModel
    {
        public string EntityType { get; set; } = string.Empty;
        public Guid EntityId { get; set; }
        public string ActivityType { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public string? Description { get; set; }
        public DateTime ActivityDate { get; set; }
        public string? CreatedByUserName { get; set; }
        public bool IsSystemGenerated { get; set; }
        public string? ReferenceEntity { get; set; }
    }
}
